File: Java/프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        int first, second;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i=0; i<scoville.length; i++) {
            pq.add(scoville[i]);
        }
        
        while (pq.size() > 1) {
            if (pq.peek() >= K) return answer;
            
            answer++;
            first = pq.poll();
            second = pq.poll();
            pq.add(first + second * 2);
        }
        
        if (pq.peek() >= K) return answer;
        return -1;
    }
}

File: Python3/프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    
    while(len(scoville) > 1):
        
        if scoville[0] >= K:
            return answer
        
        answer += 1
        first = heapq.heappop(scoville)
        second = heapq.heappop(scoville)
        heapq.heappush(scoville, first + second * 2)

    if scoville[0] >= K:
        return answer
    return -1